x/bank/keeper/msg_server.go

```diff
@@ -0,0 +1,104 @@
+package keeper
+
+import (
+	"context"
+
+	"github.com/armon/go-metrics"
+
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+type msgServer struct {
+	Keeper
+}
+
+// NewMsgServerImpl returns an implementation of the bank MsgServer interface
+// for the provided Keeper.
+func NewMsgServerImpl(keeper Keeper) types.MsgServer {
+	return &msgServer{Keeper: keeper}
+}
+
+var _ types.MsgServer = msgServer{}
```

Comment:
Because of the weird way bank has a keeper interface. Other modules should do what you're saying. And we should refactor bank in the near future anyway.

Reply:
^ cc @atheeshp

Reply:
I think in few cases we still needed this implementation, 
https://github.com/cosmos/cosmos-sdk/pull/7524/files#diff-825d33eeaa8e743955334330c753008b9af67e55c61d03b348556a246b9b4bddR94
in this scenario few of the keeper functions are colliding with the msg sever functions. wdyt @amaurymartiny, @aaronc ?

Reply:
Just use a separate struct wrapping the keeper then